elide-datastore/elide-datastore-hibernate/src/main/java/com/yahoo/elide/core/filter/HQLFilterOperation.java

```diff
@@ -178,41 +163,60 @@ public String apply(FilterExpression filterExpression, boolean prefixWithAlias)
     /**
      * Filter expression visitor which builds an HQL query.
      */
-    public class HQLQueryVisitor implements Visitor<String> {
+    public class HQLQueryVisitor implements FilterExpressionVisitor<String> {
+        public static final String TWO_NON_FILTERING_EXPRESSIONS =
+                "Cannot build a filter from two non-filtering expressions";
         private boolean prefixWithAlias;
-        private String query;
 
         public HQLQueryVisitor(boolean prefixWithAlias) {
             this.prefixWithAlias = prefixWithAlias;
         }
 
         @Override
         public String visitPredicate(FilterPredicate filterPredicate) {
-            query = apply(filterPredicate, prefixWithAlias);
-            return query;
+            return apply(filterPredicate, prefixWithAlias);
         }
 
         @Override
         public String visitAndExpression(AndFilterExpression expression) {
-            String left = expression.getLeft().accept(this);
-            String right = expression.getRight().accept(this);
-            query = "(" + left + " AND " + right + ")";
-            return query;
+            FilterExpression left = expression.getLeft();
```

Comment:
Let's make sure that the DataStore transactions only receive FilterExpressions that never have a FALSE_USER_CHECK_EXPRESSION or a NO_EVALUATION_EXPRESSION.  There will be better separation of concerns.